core/system/src/api.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2013-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <uvisor.h>
+#include "api/inc/api.h"
+#include "box_init.h"
+#include "debug.h"
+#include "halt.h"
+#include "svc.h"
+#include "unvic.h"
+#include "vmpu.h"
+#include "page_allocator.h"
+#include "thread.h"
+#include "box_init.h"
+
+#define transition_np_to_p(fn_ret, fn_name, ...) \
+    UVISOR_NAKED static \
+    fn_ret fn_name ## _transition(__VA_ARGS__) { \
+        asm volatile( \
+            "svc %[svc_id] \n" \
+            "bx lr         \n" \
+            :: [svc_id] "I" (UVISOR_SVC_ID_ ## fn_name & 0xFF) \
+        ); \
+        __builtin_unreachable(); \
+    }
+
+#define transition_p_to_p(fn_ret, fn_name, ...) \
+    UVISOR_NAKED static \
+    fn_ret fn_name ## _transition(__VA_ARGS__) { \
+        asm volatile( \
+            "ldr r12, =" #fn_name "\n" \
+            "bx r12                \n" \
+        ); \
+        __builtin_unreachable(); \
+    }
+
+/* Use this one weird trick to map functions to SVC IDs. */
+#define UVISOR_SVC_ID_debug_register_driver         UVISOR_SVC_ID_DEBUG_REGISTER_BOX
```

Comment:
The reason why I didn't do that was that if you change the function name, you also have to change the SVC IDs in many places and that would spam this PR a little.
Would you like me to add this as a separate commit though?